desci-server/src/middleware/permissions.ts

```diff
@@ -94,3 +107,69 @@ export const extractUserFromToken = async (token: string): Promise<User | null>
     });
   });
 };
+
+/**
+ * Extract API Key from IncommingRequest
+ */
+export const extractApiKey = async (request: ExpressRequest | Request) => {
+  const apiKeyHeader = request.headers['api-key'];
+  return apiKeyHeader;
+};
+
+/**
+ * Attempt to retrieve user via API key
+ */
+export const extractUserFromApiKey = async (apiKey: string, ip: string): Promise<User | null> => {
+  return new Promise(async (resolve, reject) => {
```

Comment:
Curious about the explicit promise resolve in all paths, why not just `return` since it's async? :thinking: